### PR TITLE
Add mode as `development`

### LINKED ISSUE #1560: Setup without warnings (defaults to production mode)
* Operating System: Windows 10
* Node Version: 8.9.0
* NPM Version: 6.1.0
* webpack Version: 4.24.0
* webpack-dev-server Version: 3.1.10

- [ ] This is a **bug**
- [x] This is a **modification** request

### Expected Behavior
When following the setup instructions in the readme I want to have `webpack-dev-server` running without any warnings or errors. 

### Actual Behavior
I end up with a warning because `webpack-dev-server` is in production mode. 

`WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value.`

Please note that in readme it says: _"This should be used for **development only**."_

### For Features; What is the motivation and/or use-case for the feature?
I would love one of the following:

- change the readme instructions to include the `--mode=development`. Instead of letting people get a warning because of incorrect setup, and have to figure out a solution themselves (not obvious to everyone that `webpack-dev-server` takes the same flags as `webpack`).

- This was already rejected, but I think it's highly related and deserves to be an option, so please see this as very respectfully and carefully mentioned: Revisit https://github.com/webpack/webpack-dev-server/issues/1327 (Default mode to development?), read the comments, and consider the number of 👍  vs 👎 and see if maybe it's reasonable to reconsider this request. If not, could we please have a slightly more detailed explanation to help us understand the issues? You never know, maybe someone has a different idea that would work for everyone. But to get there, they need to know the problem. 

And of course either way: Thank you for a great plugin, only getting better!

### PR DESCRIPTION
**Pull request by hiroppy**
fixes #1560 

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

yes

### Motivation / Use-Case

Separate the place to build the configuration file from `bin/webpack-dev-server.js` in order to make it easier to test.

Add `mode` flag as `development`.

### Breaking Changes

no

### Additional Info

I will increase the coverage of the test with other PRs.

### CODE REVIEW

File: bin/webpack-dev-server.js [hunk 1]
```diff
@@ -27,13 +26,14 @@ const webpack = require('webpack');
...
 
-const { colors, status, version, bonjour, defaultTo } = require('./utils');
+const { colors, status, version, bonjour } = require('./utils');
 
 const Server = require('../lib/Server');
 
 const addEntries = require('../lib/utils/addEntries');
 const createDomain = require('../lib/utils/createDomain');
 const createLogger = require('../lib/utils/createLogger');
+const createSchema = require('../lib/utils/createSchema');
```
Comment (evilebottnawi) [inline bin/webpack-dev-server.js]: Good idea, maybe better name for this are `createConfig` (avoid misleading between schema and configuration)?

File: lib/utils/createSchema.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const path = require('path');
+const { defaultTo } = require('../../bin/utils');
+
+module.exports = function processOptions(config, argv, { port }) {
```
Comment (evilebottnawi) [inline lib/utils/createSchema.js]: Same here `createConfig`

**Comment by hiroppy** [comment]
✅ rename `createSchema` to `createConfig`

File: test/__snapshots__/CreateConfig.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      "cached": "_cached",
+      "cachedAssets": "_cachedAssets",
+      "colors": "_colors",
+    },
+    "useLocalIp": true,
+    "watchContentBase": "_watchContentBase",
+    "watchOptions": undefined,
+  },
+  "entry": "./app.js",
+  "mode": "development",
```
Comment (hiroppy) [inline test/__snapshots__/CreateConfig.test.js.snap]: `mode` is added to the config.

Comment (evilebottnawi) [reply test/__snapshots__/CreateConfig.test.js.snap]: :+1:

**Comment by evilebottnawi** [comment]
Let's wait CI green and merge :+1:

**Comment by hiroppy** [comment]
Only the MacOS Node 11 fails the test.
It seems that https://github.com/webpack/webpack-dev-server/pull/1588 has recurred.

**Comment by evilebottnawi** [comment]
Very weird, because test passed before and all works fine :confused: , maybe we can find what tests fails?

**Comment by hiroppy** [comment]
@evilebottnawi CI is green 🍏  I do not know the cause😵 Thank you for rebuilding.

**Comment by evilebottnawi** [comment]
@hiroppy any ideas why coverage is dercrease?

**Comment by hiroppy** [comment]
`lib/utils` is added to codecov in this pr. https://codecov.io/gh/webpack/webpack-dev-server/compare/a52f2d7c4fe733ce4b50a560f4d6a0fe77db06f8...65956e1173840ed33c6d3716a2615e733a2c6973/tree/lib/utils

**Comment by evilebottnawi** [comment]
:+1:

**Comment by evilebottnawi** [comment]
Great job, thanks for helping :+1:

**Comment by hiroppy** [comment]
Can we release a patch version?😏

**Comment by evilebottnawi** [comment]
@hiroppy need minor release other feature will be merged already, i think we can solve couple issues and do release

**Pull request merged**